administrator/components/com_joomlaupdate/controllers/update.php

```diff
@@ -358,4 +358,53 @@ protected function _applyCredentials()
 			}
 		}
 	}
+
+	/**
+	 * Checks the admin has super administrator privileges and then proceeds with the final & cleanup steps.
+	 *
+	 * @return  array
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function confirmfinalise()
+	{
+		// Check for request forgeries
+		JSession::checkToken() or jexit(JText::_('JINVALID_TOKEN'));
```

Comment:
You cant check the token here. In previous version there is no token at this point. Only after the login submit